Kubernetes Dev Services: k3s flavour name is not properly documented

### Description

The k3s flavour needs to be passed as `k3-s` due to how SmallRye enumeration names conversions. That name is rather awkward and is not properly documented. Ideally, it should be named `k3s` but if it's not possible, the proper name should be documented accordingly.

### Implementation ideas

_No response_

confirmed the bug using:

`quarkus.kubernetes-client.devservices.flavor=k3s`

gives you:

```
Caused by: java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.kubernetes-client.devservices.flavor with the config value "k3s" threw an Exception whilst being converted Cannot convert k3s to enum class io.quarkus.kubernetes.client.runtime.KubernetesDevServicesBuildTimeConfig$Flavor
        at io.smallrye.config.SmallRyeConfig.convertValue(SmallRyeConfig.java:284)
        at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:242)
        at io.quarkus.deployment.configuration.BuildTimeConfigurationReader$ReadOperation.readConfigValue(BuildTimeConfigurationReader.java:838)
        ... 27 more
Caused by: java.lang.IllegalArgumentException: Cannot convert k3s to enum class io.quarkus.ku
```

--- a/extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/DevServicesKubernetesProcessor.java
+++ b/extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/DevServicesKubernetesProcessor.java
@@ -197,17 +197,17 @@ public class DevServicesKubernetesProcessor {
         final Supplier<RunningDevService> defaultKubernetesClusterSupplier = () -> {
             KubernetesContainer container;
             switch (config.flavor) {
-                case API_ONLY:
+                case api_only:
                     container = new ApiServerContainer(
                             config.apiVersion.map(version -> findOrElseThrow(version, ApiServerContainerVersion.class))
                                     .orElseGet(() -> latest(ApiServerContainerVersion.class)));
                     break;
-                case K3S:
+                case k3s:
                     container = new K3sContainer(
                             config.apiVersion.map(version -> findOrElseThrow(version, K3sContainerVersion.class))
                                     .orElseGet(() -> latest(K3sContainerVersion.class)));
                     break;
-                case KIND:
+                case kind:
                     container = new KindContainer(
                             config.apiVersion.map(version -> findOrElseThrow(version, KindContainerVersion.class))
                                     .orElseGet(() -> latest(KindContainerVersion.class)));

--- a/extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesDevServicesBuildTimeConfig.java
+++ b/extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesDevServicesBuildTimeConfig.java
@@ -29,7 +29,7 @@ public class KubernetesDevServicesBuildTimeConfig {
     /**
      * The flavor to use (kind, k3s or api-only). Default to api-only.
      */
-    @ConfigItem(defaultValue = "API_ONLY")
+    @ConfigItem(defaultValue = "api-only")
     public Flavor flavor;
 
     /**
@@ -70,14 +70,14 @@ public class KubernetesDevServicesBuildTimeConfig {
         /**
          * kind (needs priviledge docker)
          */
-        KIND,
+        kind,
         /**
          * k3s (needs priviledge docker)
          */
-        K3S,
+        k3s,
         /**
          * api only
          */
-        API_ONLY;
+        api_only;
     }
 }

